src/lib/pickles/wrap_domains.ml

```diff
@@ -57,12 +59,15 @@ struct
       Common.wrap_domains
         ~proofs_verified:(Nat.to_int (Nat.Add.n max_proofs_verified))
     in
-    ( if debug then
-      let res' =
-        f_debug full_signature num_choices choices_length ~feature_flags
-          ~max_proofs_verified
-      in
-      [%test_eq: Domains.t] res res' ) ;
+    (*let%map.Promise () =
+        if debug then
+          let%map.Promise res' =
+            f_debug full_signature num_choices choices_length ~feature_flags
+              ~max_proofs_verified
+          in
+          [%test_eq: Domains.t] res res'
+        else Promise.return ()
+      in*)
```

Comment:
Not sure why it's commented: if we keep the code as is, we probably need an actual comment as to why it's left that way

Reply:
will delete. the function would unnecessarily return a promise if we keep it. and that would in turn require some refactoring of the main `compile.ml` logic.

the thing is that this also means that `f_debug` is no longer simply swappable with `f` because it does return a promise. the proper refactor to fix that might be to move the promise mapping outside of `wrap_main` so that it can be synchronous here. but I'll avoid doing that before knowing that `f_debug` will even be needed